luaui/Widgets/gui_spectator_hud.lua

```diff
@@ -285,8 +287,13 @@ local function buildUnitDefs()
 	end
 
 	local function isArmyUnit(unitDefID, unitDef)
-		-- anything with a least one weapon and speed above zero is considered an army unit
-		return unitDef.weapons and (#unitDef.weapons > 0) and unitDef.speed and (unitDef.speed > 0)
+		--Spring.Echo("Spectator HUD", settings)
+		--Spring.Log("Spectator HUD", LOG.WARNING, (isCommander(unitDefID, unitDef))) 
+		-- anything with a least one weapon and speed above zero is considered an army unit	
+		-- if isCommanderArmy is false, we ignore commander units for determining army
+		if (settings.isCommanderArmy == true) then return unitDef.weapons and (#unitDef.weapons > 0) and unitDef.speed and (unitDef.speed > 0)
+		else return (not isCommander(unitDefID, unitDef)) and unitDef.weapons and (#unitDef.weapons > 0) and unitDef.speed and (unitDef.speed > 0)
```

Comment:
If you can read the code
```suggestion
		local isArmyUnit = #unitDef.weapons > 0 and unitDef.speed > 0
		return isArmyUnit and (settings.isCommanderArmy or not isCommander(unitDefID, unitDef))
```

Reply:
this does end up reading a bit more clearly, implemented.